### PR TITLE
【mlsql】【algorithm】 support grid search for linear and logistic regression learning

### PR DESCRIPTION
**Pull request by ckeys**
# What changes were proposed in this pull request?
We support grid search for linear regression and logistic regression models by group params.

In addition, I change the mapping name from 'LinearRegressionExt' to 'LinearRegression' and 'LogisticRegressor' to 'LogisticRegression'

# How was this patch tested?
- [ ] Testing done

# Are there and DOC need to update?
- [x] Doc is finished

# Spark Core Compatibility

### CODE REVIEW

**Comment by allwefantasy** [comment]
Please use following commands to combine three commits to one commit:

```shell
git rebase -i HEAD~3
```

then force push :

```shell
git push --force ckeys pengyi_master
```

**Comment by allwefantasy** [comment]
1. integrated test is missing
2. Please overwrite functions:  doc/modelType/codeExample/explainModel/auth 
3. Please also issue a doc PR in [mlsql-docs](https://github.com/allwefantasy/mlsql-docs)

File: streamingpro-it/src/test/resources/sql/simple/andie_huang_01_train_logistic_regression_model.mlsql [hunk 1]
```diff
@@ -0,0 +1,47 @@
+--%exception=java.lang.RuntimeException
```
Comment (allwefantasy) [inline streamingpro-it/src/test/resources/sql/simple/andie_huang_01_train_logistic_regression_model.mlsql]: The first two lines should be deleted.

File: streamingpro-it/src/test/resources/sql/simple/andie_huang_02_train_linear_regression_model.mlsql [hunk 2]
```diff
@@ -0,0 +1,47 @@
+--%exception=java.lang.RuntimeException
```
Comment (allwefantasy) [inline streamingpro-it/src/test/resources/sql/simple/andie_huang_02_train_linear_regression_model.mlsql]: Like the first test. Please delete the first two lines.

File: streamingpro-it/src/test/resources/sql/simple/andie_huang_02_train_linear_regression_model.mlsql [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+--%msg=all model status should be success
+--%comparator=tech.mlsql.it.IgnoreResultComparator
+!plugin app remove "mlsql-assert-2.4";
```
Comment (allwefantasy) [inline streamingpro-it/src/test/resources/sql/simple/andie_huang_02_train_linear_regression_model.mlsql]: This two lines is required?

**Review by allwefantasy — CHANGES_REQUESTED** [review]

File: streamingpro-it/src/test/scala/tech/mlsql/it/LocalBaseTestSuite.scala [hunk 4]
```diff
@@ -54,7 +55,8 @@ trait LocalBaseTestSuite extends FunSuite with SparkOperationUtil with BeforeAnd
     setupRunParams()
     copyDataToUserHome(user)
     TestManager.loadTestCase(new File(testCaseDirPath))
-    PlatformManager.getOrCreate.run(new ParamsUtil(runParams))
+    // Load built-in and external plug-ins and start PlatformManager
+    StreamingApp.main(runParams)
```
Comment (allwefantasy) [inline streamingpro-it/src/test/scala/tech/mlsql/it/LocalBaseTestSuite.scala]: You should pull latest  code first

**Review by allwefantasy — CHANGES_REQUESTED** [review]

**Comment by allwefantasy** [comment]
Please issue new docs about these two algorithms in https://github.com/allwefantasy/mlsql-docs

**Review by allwefantasy — APPROVED** [review]

**Review by lwz9103 — APPROVED** [review]

**Pull request merged**